storage/innobase/include/log0log.ic

```diff
@@ -51,6 +51,17 @@ log_block_get_flush_bit(
 	return(FALSE);
 }
 
+UNIV_INLINE
+bool log_is_in_distress()
+{
+	bool res;
+	log_mutex_enter();
+	res = 3*(log_sys.lsn - log_sys.last_checkpoint_lsn)/2 > log_sys.log_group_capacity;
+	log_mutex_exit();
+	return res;
+}
```

Comment:
I would rather not add any functions to the `.ic` files (which were renamed to `.inl` in 47e18af906f41c3b15796b8d4e6da9b744491b91). Also, I would just use the C++`inline` keyword. The macro `UNIV_INLINE` is a relic from a time when InnoDB was written in C.

That said, do we really want this function to be defined inline? Would a non-inline definition work as well? Last time I checked (admittedly, several years ago), code like `log_mutex_enter()` expanded to several hundred bytes of object code. I’d rather not duplicate that code bloat by inlining the calls.

Side note: Starting with 10.4, we use C++11, but only in later versions, all the `log_sys` fields here can be accessed without holding the mutex, thanks to `std::atomic`. But then again, the problem that is being solved here should not exist in 10.5 or later versions.

Reply:
Moved to the non-inline.